src/script/script.cpp

```diff
@@ -255,6 +256,93 @@ bool CScript::IsPayToScriptHash() const
             (*this)[22] == OP_EQUAL);
 }
 
+bool CScript::IsRedeemScriptReveal(CScript scriptpubkey) const{
+    CScript check_spk, redeemScript = scriptpubkey;
+    if (
+        // these magic numbers correspond to a typical(as of May 2021) atomicdex HLTC
+        redeemScript.size() == 110 &&
+        redeemScript[0] == OP_RETURN &&
+        redeemScript[3] == OP_IF &&
+        redeemScript[9] == OP_NOP2 &&
+        redeemScript[10] == OP_DROP &&
+        redeemScript[45] == OP_CHECKSIG &&
+        redeemScript[46] == OP_ELSE &&
+        redeemScript[47] == OP_SIZE &&
+        redeemScript[48] == 0x01 &&
+        redeemScript[49] == 0x20 &&
+        redeemScript[50] == OP_EQUALVERIFY &&
+        redeemScript[51] == OP_HASH160 &&
+        redeemScript[73] == OP_EQUALVERIFY &&
+        redeemScript[108] == OP_CHECKSIG &&
+        redeemScript[109] == OP_ENDIF
+    ) {
+        // Drop the OP_RETURN and OP_PUSHDATA1 + byte
+        redeemScript.erase(redeemScript.begin(),redeemScript.begin()+3 );
+        check_spk << OP_HASH160 << ToByteVector(CScriptID(redeemScript)) << OP_EQUAL;
+        return (check_spk == (*this));
+    }
+
+    if (
+        /*
+        these magic numbers correspond to:
+
+        OP_IF 32_bitlocktime OP_CHECKLOCKTIMEVERIFY OP_DROP pubkey OP_CHECKSIG
+        OP_ELSE OP_SIZE 0x20 OP_EQUALVERIFY
+        OP_SHA256  0000000000000000000000000000000000000000000000000000000000000000 OP_EQUALVERIFY pubkey OP_CHECKSIG OP_ENDIF
+
+        This is neccesary for coins that do not support ripemd160.
+        It is a typical HLTC with OP_SHA256 instead of OP_HASH160
+        */
+        redeemScript.size() == 122 &&
+        redeemScript[0] == OP_RETURN &&
+        redeemScript[3] == OP_IF &&
+        redeemScript[9] == OP_NOP2 &&
+        redeemScript[10] == OP_DROP &&
+        redeemScript[45] == OP_CHECKSIG &&
+        redeemScript[46] == OP_ELSE &&
+        redeemScript[47] == OP_SIZE &&
+        redeemScript[48] == 0x01 &&
+        redeemScript[49] == 0x20 &&
+        redeemScript[50] == OP_EQUALVERIFY &&
+        redeemScript[51] == OP_SHA256 &&
```

Comment:
Could you please also add `OP_HASH256` here?
```cpp
(redeemScript[51] == OP_SHA256 || redeemScript[51] == OP_HASH256`) &&
```

Reply:
Could you demonstrate a usecase for this?

Reply:
I understand it's a trivial change, but once it's in place it cannot be removed for the entire lifespan of the ARRR chain

Reply:
It's very hypothetical but we might have to support a coin that has `OP_HASH256` but doesn't support a single `OP_SHA256` for some bizarre reason :slightly_smiling_face: